packages/sdks/nextjs-sdk/test/server/authMiddleware.test.ts

```diff
@@ -79,6 +79,72 @@ describe('authMiddleware', () => {
 		expect(NextResponse.redirect).not.toHaveBeenCalled();
 	});
 
+	it('redirects unauthenticated users for private routes', async () => {
+		mockValidateJwt.mockRejectedValue(new Error('Invalid JWT'));
+		const middleware = authMiddleware({
+			privateRoutes: ['/private']
+		});
+		const mockReq = createMockNextRequest({ pathname: '/private' });
+
+		const response = await middleware(mockReq);
+		expect(NextResponse.redirect).toHaveBeenCalledWith(expect.anything());
+		expect(response).toEqual({
+			pathname: DEFAULT_PUBLIC_ROUTES.signIn
+		});
+	});
+
+	it('allows authenticated users for private routes and adds proper headers', async () => {
+		// Mock validateJwt to simulate an authenticated user
+		const authInfo = {
+			jwt: 'validJwt',
+			token: { iss: 'project-1', sub: 'user-123' }
+		};
+		mockValidateJwt.mockImplementation(() => authInfo);
+
+		const middleware = authMiddleware({
+			privateRoutes: ['/private']
+		});
+		const mockReq = createMockNextRequest({
+			pathname: '/private',
+			headers: { Authorization: 'Bearer validJwt' }
+		});
+
+		await middleware(mockReq);
+		// Expect no redirect and check if response contains session headers
+		expect(NextResponse.redirect).not.toHaveBeenCalled();
+		expect(NextResponse.next).toHaveBeenCalled();
+
+		const headersArg = (NextResponse.next as any as jest.Mock).mock.lastCall[0]
+			.request.headers;
+		expect(headersArg.get('x-descope-session')).toEqual(
+			Buffer.from(JSON.stringify(authInfo)).toString('base64')
+		);
+	});
+
+	it('allows unauthenticated users for public routes when both public and private routes are defined', async () => {
```

Comment:
also here, I think a test that ensure that if the configuration contains both public and private route - and user hit private route - the private is ignored

Reply:
I suggest a test when the sdk is configured with both private and public routes, and the user is unauthenticated and hit a route that is not private and not public 

only the public route configuration should be respected